### DISCUSSION TITLE
Biogenerator Revamp

### DISCUSSION THREAD

**Discussion by jwhitak** [vgstation-coders/vgstation13#32975] [Ideas]
# Biogenerator Changes

Imagine a world where the biogenerator gave you stuff from material besides nutriment, and the stuff is potentially more powerful or at least new. This is that world. Please discuss the pros and cons of the following proposals:

1) Rework the biogenerator and grown plants so that different plants provide different materials, and different materials are required to produce items. Motivation is to both encourage botanists to grow a variety of plants opposed to just heavy watermelons, and to provide a new source of higher end goods for the station (vs Cargo, Traders, Science, Chemistry, and Space exploration). Being a new change, it can attract players to try out botany. Should this revamp be done at all?

2) Types of materials. Examples: Wood, fiber, juice, bananas. We want to encourage a variety of plants that can be used elsewhere. Tower caps, wheat, tomatoes, bananas, that sort. What materials/general plant types should be focused on here? (I'm not giving up the banana unique material. Motivation for clown items and clown fun.) Additionally, some things can be gated behind alien (xenoarch/trader/cargo) plants, requiring interaction outside the department.

3) Types of produced goods. Biogenerators should be a source of unique things for the station that won't necessarily break the game, but be interesting. The current list can be ported, but what else would be good in this design space?

**Comment by Eneocho** [comment]
Anything that makes the biogen useful instead of a rarely used niche tool is good.
### Some assorted ideas:
* Making raw leather.
* Making banana cream pies and assorted clown gear (such as bikehorns, rubber ducks, clown apparel, etc).
* Put a cell inside the nu-biogen and it instantly charges the cell at a constant X biomass per Y power.
* Limitless production as long as there's enough biomass. For example: When making meat you at most can make 5 meat slabs each time. There could be a way to make as much meat as you can afford (with some sanity limits). Say you can afford 80 slabs, the machine takes a bit longer but it shits out 80 slabs at once instead of needing to make 5 slabs 16 times.
* Being able to feed bodies to the biogenerator if emagged, leaving only the head (it processes the meat into lots of biomass).
* Being able to make more than just monkeycubes (all basic monkey cubes are enabled by default, the dangerous ones need emagging/a sample). Say you want carpcubes, just feed an undamaged carp corpse to the machine to enable carpcube manufacture. Do this to enable the sample's mobcube. You could end up making Basilisk cubes for diamond ore profit, for example.
* Quick seed production: You feed it a seed and it consumed X biomass for every seed duplicate. QoL for faster seed manufacture instead of having to plant then take sample, if you can afford it.
* Biogen working as a faux-chem dispenser, you feed it some assorted chemicals from X Y and Z plants and it makes an appropriate medicine according to a plant-based recipe (For example: feed it rocknut, tomato and glowshrooms to make synthocarisol).
* Biogen working as a faux-protolathe. You feed it materialplants (it checks the reagent value) and adds that to its local stock of material which can be used to print somewhat advanced (not maxed research) items, like t2 parts, high/super cells, bruisekits, autoinjectors, etc. Upgraded modes would enable more expensive or complex devices, like printing a pre-made chemical EMP nade for enough uranium and iron (glowberries and rocknuts).

The faux-protolathe idea would encourage more botany requests like a medic asking for extra bruisekits since they ran out and science is unstaffed, just pop silicate citrus and rocknuts on the machine for glass+iron and make your bruisekit, or a borg asking for a supercell to replace their AA cells.

**Reply by jwhitak** [reply]
The concept I am seeing specifically is a parallel-to-protolathe, with its own unique resources based on plants. Though there are some good ideas here. Meat/skins could be a source of materials as well. Going down the list of ideas:
* I'm unsure about raw leather. I want to do it but whipcity is a thing. It could be balanced by having to throw meat or skins into the machine.
* Clown stuff, yes. That's why you must shove banana into the biogenerator. Clown stuff requires bananas.
* Not sure about this part about recharging cells, since what I want to do involves types of materials instead of just one biomass. That said, making fancy potato batteries...
* Protolathe-style "Type your amount in" fits well here.
* Feeding bodies for meat... mmm... meat... If meat is a resource, this should be enabled by default. Emagging for humanoid corpses. (But you're probably gonna butcher them anyway?)
* Multicubing is a good idea, with a default list of mobs. Potentially more, if you feed the mob to the machine. Definitely have to consider the dangerousness of this, but considering xenobio is a thing...
* This could be an item that science or the new biogenerator could make, but I don't think it'd fit with the current one. Unless seeds themselves were a resource...
* We could use plant materials to make chems. Complete with pre-done biodegradable bottles!
* And this is the idea. It's an alternate protolathe/autolathe. Instead of feeding it miner materials, you feed it plants.

**Reply by Eneocho** [reply]
If you go faux-protolathe, use current the biomass system for organics-based items, like food, clothing, soap (soap made with fat), plant chems, etc and use the dedicated materials from the materialplants for the more exotic stuff. Instead of having meat be its own resource, just equal X amount of meat to Y biomass and have it interchangeably so you can print tons of meat for some biomass cost or process tons of meat to gain that biomass back. 
Have balance-shifting items, such as whips, gated behind certain thresholds (For example: you feed the machine 10K biomass and it consumes it all "researching" biomass-compatible products it can make, enabling the next tier of items. This repeatable at higher costs for the next iteration of items.). This would let you have the whips without the risk of someone rushing them 5 minutes in.

**Reply by jwhitak** [reply]
This would miss the design goal of encouraging a variety of plants imo. It would be just big melons and the material plants with this method. 

I do note that you want to keep things interchangeable - turning plants into synthmeat for example. Might not be easily possible with an HTML menu multi material system plus the weird material types I proposed...

**Reply by jwhitak** [reply]
Perhaps materials could have a soft tier system on what they take to make as well. 
* Bare bones iron/glass
* basic mining plasma/uranium/gold/silver
* harder diamond
* rare banana/phazon.

That sort of comparison. Something like: 
* Bare bones nutriment
* basic botany wood/juice/fiber/banana
* advanced botany ore
* rare alien/exotic

**Reply by Eneocho** [reply]
I guess that would work. So long as the basic expected stuff is still available roundstart (such as meat, cheese, soap and plant chemicals)

**Comment by jwhitak** [comment]
Considering that reply threads get buried, here's the current stage of the idea:

Compare the current protolathe material system:

* Bare bones iron/glass
* basic mining plasma/uranium/gold/silver
* harder diamond
* rare banana/phazon.

To something like this:

* Bare bones nutriment
* basic botany wood/juice/fiber/banana
* advanced botany ore
* rare alien/exotic

With basic stuff like meat/cheese/soap/plantchems only requiring the most basic of materials, and more advanced stuff like clown stuff, new toys, and cube production requiring more advanced ingredients.

**Comment by poales** [comment]
I do wish the biogenerator would give botanists a more sane way to handle chemical production.
Right now the scheme is something like grind everything down, run to the condimaster, create billions of bottles in the kitchen backroom OR splice everything you want into one plant juggling disks all the way, and then you better have enough potency and room in the plant to do what you need.

It could let you store up and use reagents made by basic processing (i.e. juices, provided by the plant's morphology) and more elaborate chemicals from the plant (provided by the plant's phytochemistry). Like it would be cool to just dispense DD if you actually went to the effort of growing all the plants required, including an exotic that has inaprovaline.

If you plan on adding additional characteristic components like fiber, it would have to be additional information tied to some element of the plant, probably morphology. Many other things mentioned here, like bananas, have unique-ish chemicals that come from the right plant
